# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmValidate.h
# Commit: a7ad7e2b02c1
# Full Hash: a7ad7e2b02c11538c1f27efb67a48a4fb823ba8e
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-18 09:24:52
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Generalize table/element code for future reference-types. r=lth
#   
#   wasm::Table uses wasm::TableKind to determine the element type of the table,
#   which makes it difficult to extend tables to support more reference types. This
#   commit drops TableKind in favor of directly storing the element RefType, along
# ==============================================================================

diff -r 538920d2d777 -r a7ad7e2b02c1 js/src/wasm/WasmValidate.h
--- a/js/src/wasm/WasmValidate.h	Tue Aug 18 00:04:37 2020 +0000
+++ b/js/src/wasm/WasmValidate.h	Tue Aug 18 00:45:07 2020 +0000
@@ -224,9 +224,7 @@
   bool funcIsImport(uint32_t funcIndex) const {
     return funcIndex < funcImportGlobalDataOffsets.length();
   }
-  bool isRefSubtypeOf(ValType one, ValType two) const {
-    MOZ_ASSERT(one.isReference());
-    MOZ_ASSERT(two.isReference());
+  bool isRefSubtypeOf(RefType one, RefType two) const {
     // Anything's a subtype of itself.
     if (one == two) {
       return true;
@@ -234,7 +232,7 @@
 #if defined(ENABLE_WASM_GC)
     if (gcTypesEnabled()) {
       // Structs are subtypes of AnyRef.
-      if (isStructType(one) && two.isExternRef()) {
+      if (isStructType(one) && two.isExtern()) {
         return true;
       }
       // Struct One is a subtype of struct Two if Two is a prefix of One.
